### PR TITLE
holoviz: update all

### PR DESCRIPTION
**Pull request by Chrismarsh**
- Update all of the holoviz family of packages
- Fix a couple of missing depends bounds

✅  Depends on the `py-dask@2025` update in https://github.com/spack/spack-packages/pull/897

### CODE REVIEW

**Comment by Chrismarsh** [comment]
`py-colorcet`: @vvolkl
`py-datashader`, `py-geoviews`, `py-holoviews`: @climbfuji 
`py_param`: @haralmha

**Comment by climbfuji** [comment]
Noticing that the ci tests are failing, but otherwise the changes look good for the packages that I am concerned with.

**Comment by Chrismarsh** [comment]
Oof yes, I had to fix a dask edge case with spatialpandas after I put this through, so this PR depends on the dask update in  https://github.com/spack/spack-packages/pull/897

**Comment by Chrismarsh** [comment]
@climbfuji the required xarray fix is in. Are you able to take another look at this?

**Review by climbfuji — APPROVED** [review]
looks good to me

File: repos/spack_repo/builtin/packages/py_panel/package.py [hunk 1]
```diff
@@ -31,6 +32,7 @@ class PyPanel(PythonPackage):
     depends_on("py-bokeh@2.4.3:2.4", type=("build", "run"), when="@0.14.4")
     depends_on("py-bokeh@3.5:3.6", type=("build", "run"), when="@1.5.2:")
     depends_on("py-bokeh@3.5:3.7", type=("build", "run"), when="@1.6.3:")
+    depends_on("py-bokeh@3.7", type=("build", "run"), when="@1.7.5:")
```
Comment (climbfuji) [inline repos/spack_repo/builtin/packages/py_panel/package.py:35]: Just checking that this shouldn't be `@3.7:`?

Comment (Chrismarsh) [reply repos/spack_repo/builtin/packages/py_panel/package.py:35]: It should be `@3.7` if I've read this correctly https://github.com/holoviz/panel/blob/v1.7.5/pyproject.toml#L6

**Review by climbfuji — APPROVED** [review]

**Review by tgamblin — APPROVED** [review]
